detection_rules/rule_formatter.py

```diff
@@ -74,6 +74,31 @@ class NonformattedField(str):
     """Non-formatting class."""
 
 
+def preserve_formatting_for_fields(data: OrderedDict, fields_to_preserve: list) -> OrderedDict:
+    """Preserve formatting for specified nested fields in an action."""
+
+    def apply_preservation(target: OrderedDict, keys: list) -> None:
+        """Apply NonformattedField preservation based on keys path."""
+        for key in keys[:-1]:
+            # Iterate to the key, diving into nested dictionaries
+            if key in target and isinstance(target[key], dict):
```

Comment:
I don't think we have any list of objects that fall into this category, so should be good for this patch (and hopefully a refactor or replacement of this process soon)